usecases/user_usecase.go

```diff
@@ -6,28 +6,28 @@ import (
 
 	"github.com/checkmarble/marble-backend/models"
 	"github.com/checkmarble/marble-backend/repositories"
+	"github.com/checkmarble/marble-backend/usecases/executor_factory"
 	"github.com/checkmarble/marble-backend/usecases/security"
 	"github.com/checkmarble/marble-backend/usecases/tracking"
-	"github.com/checkmarble/marble-backend/usecases/transaction"
+
 	"github.com/cockroachdb/errors"
 )
 
 type UserUseCase struct {
 	enforceUserSecurity security.EnforceSecurityUser
-	transactionFactory  transaction.TransactionFactory_deprec
+	transactionFactory  executor_factory.TransactionFactory
 	userRepository      repositories.UserRepository
 }
 
 func (usecase *UserUseCase) AddUser(ctx context.Context, createUser models.CreateUser) (models.User, error) {
 	if err := usecase.enforceUserSecurity.CreateUser(createUser.OrganizationId); err != nil {
 		return models.User{}, err
 	}
-	createdUser, err := transaction.TransactionReturnValue_deprec(
+	createdUser, err := executor_factory.TransactionReturnValue(
 		ctx,
 		usecase.transactionFactory,
 		models.DATABASE_MARBLE_SCHEMA,
-		func(tx repositories.Transaction_deprec) (models.User, error) {
-
+		func(tx repositories.Executor) (models.User, error) {
```

Comment:
Ideally, we'll go for 
```suggestion
		func(exec repositories.Executor) (models.User, error) {
```

But I can live with that, it's painfull to replace

Reply:
well actually I left it like that on purpose, to clarify in the code within the closure that it's actually a transaction being used